### PR TITLE
Enhances Alcoholism

### PR DESCRIPTION
**Pull request by KasparoVy**
:cl:
rscadd: You can now slam drinks/bottles/flasks by click-dragging them onto yourself while it's in your active hand.
/:cl:

Chugging is interrupted by dropping the glass/whatever, being incapacitated, moving, etc.
Unless you put the drink/bottle down or any of the above happens, you'll knock back the entire thing in one go. Unless you can't/won't, know how to stop or you'll... drop. Seriously, a bottle chug FUCKS you up.

Party animals only.

### CODE REVIEW

File: code/modules/mob/living/carbon/carbon.dm [hunk 1]
```diff
@@ -1006,17 +1006,18 @@ var/list/ventcrawl_machinery = list(/obj/machinery/atmospherics/unary/vent_pump,
 
 /*TO DO - If/when stomach organs are introduced, override this at the human level sending the item to the stomach
 so that different stomachs can handle things in different ways VB*/
-/mob/living/carbon/proc/consume(var/obj/item/weapon/reagent_containers/food/toEat)
+/mob/living/carbon/proc/consume(var/obj/item/weapon/reagent_containers/food/toEat, var/bitesize_override)
+	var/bitesize = bitesize_override ? bitesize_override : toEat.bitesize
```
Comment (Fox-McCloud) [inline code/modules/mob/living/carbon/carbon.dm]: This var name mayyy get a bit confusing, what with there being a `toEat.biteze`

**Comment by Fox-McCloud** [comment]
Good heavens.

File: code/modules/mob/living/carbon/carbon.dm (hunk 1 above)
Comment (KasparoVy) [inline code/modules/mob/living/carbon/carbon.dm]: Good point, my bad. I'll make it something more unique.

**Comment by PhantasmicDream** [comment]
Just want I needed for my alcoholic Skrell.

Now I wait for the time where I can throw a glass/bottle at the wall and have it shatter.

**Comment by DarkPyrolord** [comment]
This... is exactly what I needed but never thought of.

**Comment by ktoma36** [comment]
Does this work with beakers?

**Comment by KasparoVy** [comment]
@ktoma36 Nope. Can't drink from beakers. Stuff you can drink from you can chug.

**Comment by pinatacolada** [comment]
Imagine chugging a bluespace beaker of suicider though

**Comment by KasparoVy** [comment]
Absolutely insane! Maybe there could be a bluespace bottle of some crazy liquor or something. *shrug*

**Comment by ktoma36** [comment]
I was thinking one beaker of water and one of potassium XD

**Comment by taukausanake** [comment]
We need to bluespace everything eventually. Broken or not, such advancements are only to be expected from mankind

**Comment by FalseIncarnate** [comment]
Next up on the alcoholic feature pub crawl: squatting, followed by keg-stands.

**Pull request merged**